llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java

```diff
@@ -276,11 +275,20 @@ public void unpause() {
 
   @Override
   protected Void callInternal() throws IOException, InterruptedException {
-    return ugi.doAs(new PrivilegedExceptionAction<Void>() {
-      @Override
-      public Void run() throws Exception {
-        return performDataRead();
+    return ugi.doAs((PrivilegedExceptionAction<Void>) () -> {
+      long startTime = counters.startTimeCounter();
+      try {
+        performDataRead();
+        consumer.setDone();
```

Comment:
keep it in `performDataRead`

Reply:
There are way too many return and anyone trying changing code should remember to do this correctly. This code refactors all the different call site and unifies handling error, timing and trace to just one place. Now if you see performDataRead is much much simpler.

Reply:
ok